lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1156,6 +1156,77 @@ static void helperSetter(Constant *constT, std::vector<datatype> &vec) {
   }
 }
 
+template <typename T, typename datatype>
+static void helperSetter(const glow::Constant *constT,
+                         std::vector<datatype> &vec) {
+  auto constH = constT->getPayload().getHandle<T>();
+  for (size_t i = 0; i < constH.size(); ++i) {
+    vec.push_back(constH.at({i}));
+  }
+}
+
+Error ONNXModelLoader::loadRange(const ONNX_NAMESPACE::NodeProto &op,
+                                 ArgumentDictionaryTy &dict) {
+  const Constant *constT;
+  ASSIGN_VALUE_OR_RETURN_ERR(constT, getConstantByName(op.input(0)));
+
+  auto glowType = constT->getElementType();
+  Tensor T;
+  if (glowType == ElemKind::Int64ITy) {
+    int64_t start = constT->getPayload().getHandle<int64_t>().raw(0);
+
+    ASSIGN_VALUE_OR_RETURN_ERR(constT, getConstantByName(op.input(1)));
+    int64_t limit = constT->getPayload().getHandle<int64_t>().raw(0);
+
+    ASSIGN_VALUE_OR_RETURN_ERR(constT, getConstantByName(op.input(2)));
+    int64_t delta = constT->getPayload().getHandle<int64_t>().raw(0);
+
+    std::vector<int64_t> rangeValue;
+    if (limit > start) {
+      RETURN_ERR_IF_NOT(delta > 0, "delta should be positive");
+      for (auto i = start; i < limit; i += delta)
+        rangeValue.push_back(i);
+    } else if (limit < start) {
+      RETURN_ERR_IF_NOT(delta < 0, "delta should be negative");
```

Comment:
We don't have this check in the int32 version -- shouldn't we have it for both?

Reply:
Added a template helper function.